### PR TITLE
DeformableConvolution specification refactoring.

### PR DESCRIPTION
**Pull request by jdanieck**
DeformableConvolution specification refactored. I used [MXNet](https://mxnet.apache.org/versions/1.7.0/api/python/docs/api/contrib/ndarray/index.html#mxnet.contrib.ndarray.DeformableConvolution) as a reference. Both MXNet and CPU plugin supports only 2D so I stated this explicitly in specification.

Details:
- Changed input & output tensor types to 'any floating point'
- Added short description
- Extended detailed description
- Added examples
- Added description of inputs & output rank and layout

### CODE REVIEW

File: docs/ops/convolution/DeformableConvolution_1.md [hunk 1]
```diff
@@ -4,85 +4,180 @@
...
-  * **Default value**: 1
+  * **Default value**: `1`
   * **Required**: *no*
 
 **Inputs**:
 
-*   **1**: Input tensor of rank 3 or greater. Required.
+*   **1**: Input tensor of type *T* and rank 3, 4 or 5. Layout is NCZYX (number of batches, number of channels, spatial axes Z, Y, X). Required.
+
+*   **2**: Deformable values tensor of type *T* and rank 3, 4 or 5. Layout is *TBD*. Required.
```
Comment (jdanieck) [inline docs/ops/convolution/DeformableConvolution_1.md:80]: @lazarevevgeny @dmitry-gorokhov can you help me with understanding 'deformable values' input and what its layout should be?

Comment (jdanieck) [reply docs/ops/convolution/DeformableConvolution_1.md:80]: @iimironov maybe you can help here?

File: docs/ops/convolution/DeformableConvolution_1.md [hunk 2]
```diff
@@ -4,85 +4,180 @@
...
+**Types**:
+
+* *T*: Any floating point type.
+ 
 **Example**
 
+1D Convolution
+```xml
+<layer type="DeformableConvolution" ...>
+    <data dilations="1" pads_begin="0" pads_end="0" strides="2" auto_pad="valid" group="1" deformable_group="1"/>
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_1.md:102]: Why all the samples have 3 inputs? According to the specification there should be exactly 3 inputs.

File: docs/ops/convolution/DeformableConvolution_1.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/ops/convolution/DeformableConvolution_1.md:80]: MO should not modify the layout of weights in the DeformableConvolution (@iimironov , could you double check that?). So the layout should be as described https://github.com/apache/incubator-mxnet/blob/master/src/operator/deformable_convolution.cc#L64-L68

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/convolution/DeformableConvolution_1.md (hunk 1 above)
Comment (jdanieck) [reply docs/ops/convolution/DeformableConvolution_1.md]: Fixed.

File: docs/ops/convolution/DeformableConvolution_1.md (hunk 2 above)
Comment (jdanieck) [reply docs/ops/convolution/DeformableConvolution_1.md]: Yes it was a mistake. Fixed. I used MXNet framework to verify that we can execute DeformableConvolution with dimensions provided in examples.

**Review by lazarevevgeny — APPROVED** [review]

**Comment by jdanieck** [comment]
@yury-intel as you were designated to represent CPU plugin team on the topic of DeformableConvolution, please have a look at this PR and confirm that it is aligned with CPU plugin implementation. I'd like to merge it.

**Review by yury-intel — APPROVED** [review]
Looks correct

**Comment by jdanieck** [comment]
@lazarevevgeny looks ready to merge

**Pull request merged**